src-tauri/src/ingestion/mod.rs

```diff
@@ -1,16 +1,38 @@
-use std::path::PathBuf;
+use std::{
+    fs::{self, DirEntry},
+    path::PathBuf,
+};
+
+use serde::{de::DeserializeOwned, Serialize};
 
 pub mod pf2e;
 
 /// Trait for ingesting data from a source
-pub trait Ingest {
-    /// The output type of the ingestion.
-    type Output;
+pub trait Ingest: Serialize + Named + Sized + DeserializeOwned {
```

Comment:
This is a good way to make this generic! In the future it will cahnge, so good job commenting what everything does!